### PR TITLE
move hover/select styling to rowRender

### PR DESCRIPTION
**Pull request by bucko13**
Changes based on comments in [PR #39](https://github.com/bpanel-org/bpanel-ui/pull/39)

### CODE REVIEW

**Comment by pinheadmz** [comment]
Haven't tested yet, but doesn't the `tableRowStyle` function override the hover/selected classes? I think that was my issue before. So even though `rowRenderer` parses hover and selected, it decorates the rows by using class names. Then `tableRowStyle` runs, producing INLINE CSS to decorate the alternating row pattern. My result was getting the alternating background, but the hover/select didn;t show through. Anyway, gunna try to test this out now.

**Comment by pinheadmz** [comment]
Yeah ok _"It's what I feared!"_  The `cursor:pointer` is getting through, because the inline style from `tableRowStyle` only adds a background color. But that bg color **does** override the color set by the classes, which are switched by hover and select. That's why I was going the route of doing everything as inline styles, if you want theme plugins to be able to style rows by arbitrary function, it's going to end up being inline, which will override anything set by a class.

Original:
before mov

After merging this PR:
after mov

**Comment by bucko13** [comment]
Ah ok. I see what you mean. Interestingly, I was testing this out on the simple-wallet selectable implementation and it seemed to work although after you pointed this out I noticed that it only worked on _odd_ numbered rows. Changed my theme and turned out that behavior was also just for bdark... So yeah, this is a tough one 😅. 

This feels to me like something that should be in the class names and that we should avoid using the classes in a way that they aren't really intended to (i.e. as in-line styles). One possible solution would have been to get rid of the `tableRowStyle` from the theme all together, but this messed up the odd/even styling (and the header). This could be fixed by adding odd/even as a class but I don't feel that quite makes sense as a stand alone class since it feels a little too specific to the implementation, but hover/selected should be. 

I managed to get this working by just adding `!important` to the `backgroundColor` for the `hoverRow` and `selectedRow` classes in the theme object. The downside of this is that if someone does a custom theme where they replace these classes and _don't_ add `!important`, their theme won't render correctly, but that should be immediately obvious when implementing. On the positive side, this approach allows the classes to be used as intended, the table will update appropriately if the `themeColors` change, and it feels like it simplifies the logic of the Table code a little bit by moving the styling logic back to the `rowRenderer`.

I also added logic to the `rowRenderer` to apply header styles if it's the header row. For some reason, background color still needs to be applied via the `tableRowStyle` function, might be able to play around with this a bit more but not super important I guess.

So to summarize, this would mean inline styles are, by default, only for applying custom styles to specific rows based on some condition (e.g. odd or even by default). The classes are used for everything else: header, selected, hovered, and hoverExpandable (except for header's background color 🤷‍♂️ ). A custom theme developer can change the hover/selectable colors by changing the palette in `themeColors`, overriding the table css theme including using `!important` for selectable and hoverable rows, OR doing manual logic in a custom `tableRowStyle`.

What do you think about this? Does it make sense or still preferable in your opinion to pull out the themeConfig? 

Here's the preview of it working with bdark at least!
screen recording 2018-11-23 at 8 41 25 pm

**Comment by bucko13** [comment]
wow, that comment got longer than I expected/intended 😅

File: src/components/Table/Table.js [hunk 1]
```diff
@@ -111,7 +111,7 @@ class Table extends PureComponent {
 
   selectRow({ index }) {
     const { expandedHeight } = this.props;
-    const { selectedIndex } = this.state;
+    const { selectedIndex, hoverIndex } = this.state;
```
Comment (pinheadmz) [inline src/components/Table/Table.js:114]: do we need `hoverindex` here? It's not used in this function

Comment (bucko13) [reply src/components/Table/Table.js:114]: Ah, no. Doesn't look like it. Wonder if this got left from something else I was trying 🤔. Linter should have caught this too. This is why we need CI! (though I guess it wouldn't have mattered for our weird incestuous PR branch on a PR branch 😬)
